app/controllers/friend_requests_controller.rb

```diff
@@ -0,0 +1,106 @@
+class FriendRequestsController < ApplicationController
+  before_action :set_friend_request, only: [:accept, :decline, :destroy]
+
+  def index
+    @sent_requests = current_user.friend_requests_sent.pending
+    @received_requests = current_user.friend_requests_received.pending
+  end
+
+  def create
+    @friend_request = FriendRequest.find_or_initialize_by(sender_id: current_user.id,
+                                                          receiver_id: params[:friend_request][:receiver_id])
+    @friend_request.status = 0
+    respond_to do |format|
+      if @friend_request.save
+        format.html { redirect_back fallback_location: root_path,
+                      success: "Friend request sent succesfuly" }
+        format.json { render json: @friend_request, status: :created,
+                      location: @friend_request }
+        format.js
+      else
+        format.html { redirect_back fallback_location: root_path,
+                      danger: "Something went wrong"  }
+        format.json { render json: @friend_request.errors,
+                      status: :unprocessable_entity }
+        format.js
+      end
+    end
+  end
+
+  def accept
```

Comment:
I tried to implement it with another way: skinny model, skinny controller, fat service :) Some of the functionalities look so weird when we put in the model. I'm not against completely putting some functionalities on the model. But, you're right, accept & decline must be service names, not an action names :) If you have time, can you look quickly and give me a short response because I need to convert some of my partials through the changes. Tests are ok for now. The last thing, I don't like the usage of enum, I want to refactor it with type object, what do you think about it?